tools/c7n_azure/tests/test_template_utils.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2015-2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from azure_common import BaseTest
+from c7n_azure.session import Session
+from c7n_azure.template_utils import TemplateUtilities
+
+
+class TemplateUtilsTest(BaseTest):
+    def setUp(self):
+        super(TemplateUtilsTest, self).setUp()
+        self.template_util = TemplateUtilities()
+
+    def test_create_resource_group(self):
+        group_name = 'cloud-custodian-test'
+        self.template_util.create_resource_group(group_name, {'location': 'West US 2'})
+        s = Session()
+        #: :type: azure.mgmt.resource.ResourceManagementClient
+        client = s.client('azure.mgmt.resource.ResourceManagementClient')
+        resource_group = client.resource_groups.get(group_name)
+        self.assertIsNotNone(resource_group)
```

Comment:
looks like this creates with static name but does not delete?  Will be hard for next person to re-record without going to manually delete, also two people working in our sub would conflict.  Might consider adding a random number or something to the group name at test setup at least. Also `tearDown()` is available.

Reply:
I'll run tearDown() and delete the resource group!

Reply:
Changed my mind - will just delete in this test since tearDown() is executed after ever unit test and only one test requires a network call in this test file.